### ISSUE TITLE
[java] CloseResource not working on PreparedStatement

### ISSUE THREAD

**Issue by regrog** [pmd/pmd#1665]
**Affects PMD Version:** 
6.11.0

**Rule:**
[CloseResource](https://pmd.github.io/pmd-6.11.0/pmd_rules_java_errorprone.html#closeresource)

**Description:**
I found in my code a `java.sql.PreparedStatement` not closed but I expected PMD to notify me about that.
If I change it to a `java.sql.Statement`, PMD notifies me.

**Code Sample demonstrating the issue:**

It's an eclipse plugin but you can test it using command line: `./gradlew PmdMain`

[projectA.zip](https://github.com/pmd/pmd/files/2879255/projectA.zip)

**Running PMD through:** *Gradle 5.2*

java 8

**Comment by adangel** [comment]
Hi @regrog,
thanks for reporting!

The rule can be configured and by default only looks for the following classes:

* java.sql.Connection
* java.sql.Statement
* java.sql.ResultSet

so, you would need to change the types properties, e.g.

```xml
<rule ref="category/java/errorprone.xml/CloseResource">
    <properties>
        <property name="types" value="java.sql.Connection,java.sql.Statement,java.sql.ResultSet,java.sql.PreparedStatement" />
    </properties>
</rule>
```

There is a related issue for this rule: #1000 and a PR is progress #1539 . This would enable type resolution, so that `PreparedStatement` would indeed be identified as a subtype of `Statement`. The PR would enable the rule to check for more than just SQL resources, since all resources should be closed. But this yields at the moment a couple of false positives.

Let me know, if changing the property "types" works for you.

**Comment by regrog** [comment]
I'll try this tomorrow, thanks!
Reading the documentation I didn't realize that I could add `java.sql.PreparedStatement` in the properties, great!

**Comment by regrog** [comment]
I'm a bit late...
You solution works perfectly!
